plugins/de.cau.cs.kieler.klighd/src/de/cau/cs/kieler/klighd/filtering/BinaryConnective.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * KIELER - Kiel Integrated Environment for Layout Eclipse RichClient
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ * 
+ * Copyright 2022 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package de.cau.cs.kieler.klighd.filtering;
+
+/**
+ * A binary connective C takes two filter rules R1 and R2 as operands and creates the new rule R1 C R2.
+ * @author mka
+ *
+ */
+public abstract class BinaryConnective extends SemanticFilterRule {
+    
+    protected String name;
+    protected SemanticFilterRule leftOperand;
+    protected SemanticFilterRule rightOperand;
+    
+    /**
+     * {@inheritDoc}
+     */
+    public BinaryConnective() {
+        
+    }
+    
+    /**
+     * {@inheritDoc}
+     */
+    public BinaryConnective(String ruleName) {
+        super(ruleName);
+    }
+    
+    /**
+     * Returns a string representation of the rule of the form 'C(R1, R2)'.
+     * @return the rule string
+     */
+    public String getString() {
+        return name + "(" + leftOperand + ", " + rightOperand + ")";
+    }
```

Comment:
What is this used for? Why does it not override the default toString method?

Reply:
should be toString